### DISCUSSION TITLE
Discussing Intermittent Loss of sb-*-code-verifier Cookie in Supabase Social Login

### DISCUSSION THREAD

**Discussion by egoing** [supabase/supabase#22707] [Questions]
I am implementing social login features using Supabase. During the login process, a `sb-*-code-verifier` cookie is stored before redirection to the authentication server. However, we are encountering an issue where this cookie is intermittently lost. This problem occurs sporadically.

To mitigate this issue, we are currently monitoring and automatically resetting the cookie when it disappears using the following method:

```typescript
createBrowserClient(
  ...
  {
    cookies: {
      set(name, value, options) {
        setCookie(name, value, options);
        const suffix = '-code-verifier';
        if (name.endsWith(suffix)) {
          setInterval(() => {
            if (!document.cookie.includes(suffix)) {
              setCookie(name, value, options);
            }
          }, 1);
        }
      }
    }
  }
);
```

This approach is a temporary solution, and we are seeking to identify the cause and find a more fundamental resolution. If you have experienced similar issues or have suggestions or solutions for this problem, please share them. Your input would be highly appreciated.

**Comment by j4w8n** [comment]
Can you explain the real issue you're experiencing?

That cookie correctly gets removed by the `exchangeCodeForSession` method.
